components/process.rst

```diff
@@ -128,6 +128,28 @@ environment variables using the second argument of the ``run()``,
     // On both Unix-like and Windows
     $process->run(null, ['MESSAGE' => 'Something to output']);
 
+Setting Environment Variables for Processes
+-------------------------------------------
+
+The constructor of the :class:`Symfony\\Component\\Process\\Process` class and
+all its methods related to executing processes (``run()``, ``mustRun()``,
+``start()``, etc.) allow passing an array of environment variables to set before
+running the process::
+
+    $process = new Process(['...'], null, ['ENV_VAR_NAME' => 'value']);
+    $process = Process::fromShellCommandline('...', null, ['ENV_VAR_NAME' => 'value']);
+    $process->run(null, ['ENV_VAR_NAME' => 'value']);
+    // ...
+
+These environment variables are automatically inherited by child processes. You
```

Comment:
I need some help trying to understand this in practice. Imagine this command:

```php
$process = new Process(['foo'], null, ['APP_ENV' => 'test']);
```

`foo` command is run with `APP_ENV=test` and if `foo` starts new processes or runs commands, they will use `APP_ENV=test` too. OK.

But here we are saying, if you want to prevent that, set the env var to `false`:

```php
$process = new Process(['foo'], null, ['APP_ENV' => false]);
```

OK, so child processes now don't use `APP_ENV=test` ... but the main `foo` command doesn't either, right?

Question: how can I define an env var for a command ... and prevent defining it for its child processes? Thanks.

Reply:
A running `Process` instance *is* a child process. We don't make a difference between a child process and its sub-children. That's not out responsibility at all.

Reply:
Now I understand. Thanks!